packages/filesystem/src/browser/filesystem-preferences.ts

```diff
@@ -78,6 +78,12 @@ These have precedence over the default associations of the languages installed.'
             type: 'number',
             default: MAX_FILE_SIZE_MB,
             markdownDescription: 'Controls the max file size in MB which is possible to open.'
+        },
+        'files.trimTrailingWhitespace': {
+            'type': 'boolean',
+            'default': false,
+            'description': 'When enabled, will trim trailing whitespace when saving a file.',
+            'scope': 'language-overridable'
```

Comment:
I see that we state the scope is `language-overridable` however the following does not work for me:

```json
"[markdown]":{
  "files.trimTrailingWhitespace": true
}
```

With this preference configuration, `markdown` files are properly trimmed but so are other files.
I believe it is the bug that Kenneth also experienced: https://github.com/eclipse-theia/theia/pull/8742#issuecomment-726163280.

Reply:
I'll switch this over to `Folder`, since that seems to entail overridability per language. An alternative - that would also resolve potential issues with other preferences using the `language-overridable` string scope - would be to add `language-overridable` to this function as an alias of `Folder`:
https://github.com/eclipse-theia/theia/blob/cd85908d3bb5af06477ead7168110f2306a42145/packages/core/src/common/preferences/preference-scope.ts#L55-L64

Any preference (😉) between the two approaches?

Reply:
For now, I've set `'language-overridable'` to be an alias of `PreferenceScope.Folder`, and that seems to allow for language overrides.